src/Tools/Source/RunTests/ProcessTestExecutor.cs

```diff
@@ -25,52 +26,68 @@ internal ProcessTestExecutor(TestExecutionOptions options)
             Options = options;
         }
 
-        public string GetCommandLine(AssemblyInfo assemblyInfo)
-        {
-            return $"{Options.XunitPath} {GetCommandLineArguments(assemblyInfo)}";
-        }
-
         public string GetCommandLineArguments(AssemblyInfo assemblyInfo)
         {
             var assemblyName = Path.GetFileName(assemblyInfo.AssemblyPath);
-            var resultsFilePath = GetResultsFilePath(assemblyInfo);
-            var xmlResultsFilePath = Path.ChangeExtension(resultsFilePath, ".xml");
-            var htmlResultsFilePath = Path.ChangeExtension(resultsFilePath, ".html");
 
             var builder = new StringBuilder();
-            builder.AppendFormat(@"""{0}""", assemblyInfo.AssemblyPath);
-            builder.AppendFormat(@" {0}", assemblyInfo.ExtraArguments);
-            builder.AppendFormat($@" -xml ""{xmlResultsFilePath}""");
+            builder.Append($@"test");
+            builder.Append($@" ""{assemblyInfo.AssemblyPath}""");
+            var typeInfoList = assemblyInfo.PartitionInfo.TypeInfoList;
+            if (typeInfoList.Length > 0 || !string.IsNullOrWhiteSpace(Options.Trait) || !string.IsNullOrWhiteSpace(Options.NoTrait))
+            {
+                builder.Append(" --filter ");
+                var any = false;
+                foreach (var typeInfo in typeInfoList)
+                {
+                    MaybeAddSeparator();
+                    builder.Append(typeInfo.FullName);
+                }
 
-            if (Options.IncludeHtml)
-                builder.AppendFormat($@" -html ""{htmlResultsFilePath}""");
+                if (Options.Trait is object)
+                {
+                    foreach (var trait in Options.Trait.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries))
+                    {
+                        MaybeAddSeparator();
+                        builder.Append($"Trait={trait}");
+                    }
+                }
 
-            builder.Append(" -noshadow -verbose");
+                if (Options.NoTrait is object)
+                {
+                    foreach (var trait in Options.NoTrait.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries))
+                    {
+                        MaybeAddSeparator('&');
+                        builder.Append($"Trait!~{trait}");
+                    }
+                }
 
-            if (!string.IsNullOrWhiteSpace(Options.Trait))
-            {
-                var traits = Options.Trait.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
-                foreach (var trait in traits)
+                void MaybeAddSeparator(char separator = '|')
```

Comment:
Please use the camelCase local function naming convention.

Reply:
I believe the naming style for this directory is configured for PascalCase

Reply:
This directory is PascalCased yes. 

We need to flip the repo to be consistent here now that .NET runtime came out with their recomendations so we can avoid this in the future.